docs/source/torch.compiler_aot_inductor.rst

```diff
@@ -0,0 +1,184 @@
+
+
+AOTInductor: Ahead-Of-Time Compilation for Torch.Export-ed Models
+=================================================================
+
+.. warning::
+
+    AOTInductor and its related features are in prototype status and are
+    subject to backwards compatibility breaking changes.
+
+AOTInductor is a specialized version of
+`TorchInductor <https://dev-discuss.pytorch.org/t/torchinductor-a-pytorch-native-compiler-with-define-by-run-ir-and-symbolic-shapes/747>`__
+, designed to process exported PyTorch models, optimize them, and produce shared libraries as well
+as other relevant artifacts.
+These compiled artifacts are specifically crafted for deployment in non-Python environments,
+which are frequently employed for inference deployments on the server side.
+
+In this tutorial, you will gain insight into the process of taking a PyTorch model, exporting it,
+compiling it into a shared library, and conducting model predictions using C++.
+
+
+Model Compilation
+---------------------------
+
+Using AOTInductor, you can still author the model in Python. The following example demonstrates how to
+invoke ``aot_compile`` to transform the model into a shared library.
+
+.. note::
+
+   To execute the following code, it's essential to have a CUDA-enabled device on your machine.
+   If you do not possess a GPU, you can simply omit the ``.to(device="cuda")`` code within the snippet
+   below. In such a case, the script will compile the model code into a shared library that is optimized
+   for CPU execution.
+
+.. code-block:: python
+
+    import torch
+
+    class Model(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.fc1 = torch.nn.Linear(10, 16)
+            self.relu = torch.nn.ReLU()
+            self.fc2 = torch.nn.Linear(16, 1)
+            self.sigmoid = torch.nn.Sigmoid()
+
+        def forward(self, x):
+            x = self.fc1(x)
+            x = self.relu(x)
+            x = self.fc2(x)
+            x = self.sigmoid(x)
+            return x
+
+    with torch.no_grad():
+        m = Model().to("cuda")
+        example_inputs=(torch.randn(8, 10).to("cuda"),)
+        batch_dim = torch.export.Dim("batch", min=1, max=1024)
+        dynamic_shapes = {"x": {0: batch_dim}}
+        so_path = torch._export.aot_compile(m, example_inputs, dynamic_shapes=dynamic_shapes)
```

Comment:
Feels weird to get a path back? Why is that not an input to specify the path?

Reply:
Specifying an input can be done through additional options. I was debating with myself whether to do that here, mostly for the reason you are commenting later for using a file to pass the path between Python and C++. It will be some kind of hard-coded either way, but maybe you are right that specifying a path here feels more natural. I will change it.

Reply:
tbh without reading the variable names, what I though happened here is that your functions returns a custom python object representing the binary and the open/write below was actually writing the .so.

I realized while backtracking that the open was not writing the .so haha